routes/auth.js

```diff
@@ -0,0 +1,78 @@
+const router = require('express').Router();
+const User = require('../model/User');
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+const {
+  registerValidation,
+  loginValidation
+} = require('../validation');
+
+//REGISTRATION ROUTE
+router.post('/register', async (req, res) => {
+
+  //VALIDATE
+  const {
+    error
+  } = registerValidation(req.body);
+  if (error) {
+    return res.status(400).send(error.details[0].message);
+  }
+
+  //CHECK FOR EXISTING email
+  const emailExists = await User.findOne({
```

Comment:
Why not validate this in the `registerValidation` function?

Reply:
Because registerValidation just returns if the credentials entered are valid or not using the @hapi/joi package whereaes this email exists finds in the database and returns.

Reply:
> Because registerValidation just returns if the credentials entered are valid or not using the @hapi/joi package whereaes this email exists finds in the database and returns.

Okay, let it be